### ISSUE TITLE
CA1000 enhancement: Don't report for static local functions

### ISSUE THREAD

**Issue by Youssef1313** [dotnet/roslyn-analyzers#4510]
### Analyzer

**Diagnostic ID**: [CA1000](https://docs.microsoft.com/dotnet/fundamentals/code-analysis/quality-rules/ca10000)

### Describe the improvement

Don't report a diagnostic for local functions.

The goal of this rule is to avoid specifying the generic type when calling a static method. The example from the documentation discourages things such as:

```csharp
// Static method in a generic type.
GenericType<int>.StaticMethod();
```

However, this doesn't apply to static local functions.

For example:

```csharp
using System;

public class C<T>
{
    public void M() 
    {
        PrintHelloWorldLocal();

        static void PrintHelloWorldLocal() => Console.WriteLine("Hello, world");
    }
}
```

The local function is called without the need to specify a generic type argument.

**Comment by Youssef1313** [comment]
I caught this case by chance when I came across the following suppressions:

https://github.com/dotnet/roslyn-analyzers/blob/204889dc30c3a1b5f321419fe9950660fbfd659d/src/Utilities/Compiler/BoundedCache.cs#L3

**Comment by Youssef1313** [comment]
The above suppression is unnecessary. The rule is working as expected.
